### PR TITLE
Adding network troubleshooting doc.

### PR DESCRIPTION
**Pull request by mpinol**
Start of doc to add TCP Connector and Endpoint troubleshooting steps.

### CODE REVIEW

File: tutorials/ros_unity_integration/network_troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+```
+
+```
+SocketException: Connection refused
+```
+
+Confirm that:
+
+- `server_endpoint` is running
+- Can ping ROS machine from Unity machine
```
Comment (hyounesy) [inline tutorials/ros_unity_integration/network_troubleshooting.md]: code snippet? e.g. `rosnode list` ?

**Review by hyounesy — COMMENTED** [review]
I think the document addresses the issues that I have encountered with.  two overall comments:
- I feel the structure can be further improved. perhaps the sections need to be organized, and steps numbered, to make it easier to follow? and perhaps a high level TOC at the top?
- adding a few code snippets for the indicated items.

File: tutorials/ros_unity_integration/network.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+## Definitions
+
+`ROS_IP` : The IP address of the machine, VM, or container running ROS.
+
+`UNITY_IP` : The IP address of the machine running Unity.
+
+> It is possible to set both of these variables on the machines running Unity and ROS. The specifics of where and why each of these settings will be described below.
+
+On the ROS machine these settings are set as a rosparam and will typically be set in a launch file like [this](https://github.com/Unity-Technologies/Unity-Robotics-Hub/blob/main/tutorials/ros_packages/robotics_demo/launch/robo_demo.launch) or in a [param file](https://github.com/Unity-Technologies/Unity-Robotics-Hub/blob/main/tutorials/pick_and_place/ROS/src/niryo_moveit/config/params.yaml) loaded by a launch file like [this](https://github.com/Unity-Technologies/Unity-Robotics-Hub/blob/main/tutorials/pick_and_place/ROS/src/niryo_moveit/launch/part_3.launch#L2).
```
Comment (at669) [inline tutorials/ros_unity_integration/network.md:11]: suggestion: maybe calling out that if the param file isn't loaded by a roslaunch, they'll need to call `rosparam load`?

**Comment by WalterRainey** [comment]
I ordered 4 Lamborghinis back in the eighties and I had problems with
foreign trade. I sued and won the whole world and a racist CIA agent sold
my government. I need help getting my money because she started Social
Security with my program and I didn't get any compensation.

File: tutorials/ros_unity_integration/network.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+![](images/troubleshoot-docker-unity.png)
+
+## Explicitly setting `UNITY_IP`
+
+The `UNITY_IP` can be set in two different places.
+
+1. If set on the ROS side as a rosparam, the `server_endpoint` will only use this IP to send messages to Unity.
+1. If set on the Unity side as the `Override Unity IP Address`, the `UNITY_IP` on the ROS side will be set to this value during the initial handshake between ROS and Unity once play is pressed in the Editor.
+
+> If the `UNITY_IP` is not set in either of these places the IP that makes the first connection to ROS during the initial handshake will be used.
```
Comment (sarah-gibson) [inline tutorials/ros_unity_integration/network.md:51]: For readability: "...places, then the IP..."

File: tutorials/ros_unity_integration/network.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+# Troubleshooting
+
+## Where Does Communication Break Down
+
+### 1. Can Unity send messages to ROS?
+
+When pressing play in the Editor a handshake message is sent from Unity to ROS.
```
Comment (sarah-gibson) [inline tutorials/ros_unity_integration/network.md:59]: For readability: "When play is pressed in the Editor, a handshake..."

File: tutorials/ros_unity_integration/network.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+If Unity can communicate with ROS the following message should be printed to the console screen running the `server_endpoint.py` script.
```
Comment (sarah-gibson) [inline tutorials/ros_unity_integration/network.md:61]: insert comma between "ROS" and "the following"

**Review by sarah-gibson — APPROVED** [review]

**Pull request merged**